MC Config - start work to make it a complete overview (#2734)
PX4/PX4-user_guide: en/config_mc/README.md

- [Flight Controller Peripherals](../peripherals/README.md) - Setup specific sensors, optional sensors, actuators, and so on.

:::note
Barometers don't need calibration, and if you've mounted the flight controller level, there isn't much value in a [Level Horizon](../config/level_horizon_calibration.md) calibration either.

Obviously you don't need to calibrate or configure anything that isn't present or used by PX4.
For multicopter, that includes [airspeed sensors](../config/airspeed.md), which are not used for flight control.
:::

## GPS Setup

<!-- TBD 

Note that it is plugnplay except for RTK. Worth saying.

-->

## Manual Control Setup

<!-- note that you will have one or the other -->

Radio Control:

- [Radio Controller (RC) Setup](../config/radio.md)
- [Flight Mode Configuration](../config/flight_mode.md)
  
Joystick/GamePad:

- [Joystick Setup](../config/joystick.md)

## Safety Configuration

- [Battery/Power Module Setup](../config/battery.md)
- [Safety Configuration (Failsafes)](../config/safety.md)

## Tuning

<!-- 
- Explain what you have to tune on PX4, what you can tune, and what each topic covers

> - I expect we should start with an exhaustive list of the tuning you could want to do - such as position tuning, etc. Do we have one?

Author: Note, so here we have rate tuning (autotune), rate/position etc in the basic manual, +airmode tuning in the advanced, + setpoint tuning. Also latency/filters. Is there a complete list of the tuning that someone might want to do "of this type"?

Reviewer: I think we need to organize the tuning pages a bit more, but for now I find it okay to just link them all here.

Author: Yes. For me the most important things right now are to:
- Clarify when Autotune can be used (and what to do if not)
- Have a list of all the tuning we might reasonable want to do, so we can design our tuning approach.

They don't include the output mappings, unless the frame is pre-wired and non modifiable (so you would have to set this yourself in the next step).
  
The frames named with "Generic" generally just define the vehicle type, number of rotors, and "placeholder" rotor positions; almost everything needs to be configured.
:::

The next step is to define your vehicle [geometry](../config/actuators.md#motor-geometry-multicopter) (the number of motors and their relative positions) and [assign those motors](../config/actuators.md#actuator-outputs) to the physical outputs that they are wired to on your flight controller (both of these are covered in [Actuator Configuration and Testing](../config/actuators.md)).

If using PWM ESCs and OneShot ESCs (but not DShot and DroneCAN/Cyphal ESC) you should then perform [ESC Calibration](../advanced_config/esc_calibration.md) before proceeding to [Motor Configuration](../config/actuators.md#motor-configuration).
This ensures that all ESC provide exactly the same output for a given input (ideally we'd calibrate ESCs first, but you can't calibrate your ESCs until outputs are mapped).

> The final step is [Motor Configuration](../config/actuators.md#motor-configuration), which sets the directions of motors, and their disarmed, armed and high limits.

Reviewer: ```suggestion
The final step is [Motor Configuration](../config/actuators.md#motor-configuration), which for PWM ESCs sets the disarmed, low and high PWM limits.
```

Reviewer: The direction cannot be configured for normal PWM ESC.

Author: Yes, but my understanding is that you still need to set the limits even for non-PWM ESC. Please correct me if I'm wrong. Assuming I am not, how about:

```suggestion
The final step is [Motor Configuration](../config/actuators.md#motor-configuration), which sets the motor input values for  disarmed, low and high speed.
For CAN and D-Shot it can also set motor direction.
```

Reviewer: I'm not sure. Also the GUI implies that setting the direction there is not possible. 
![image](https://github.com/PX4/PX4-user_guide/assets/26798987/262c2943-093e-4459-8390-65707c0aeb96)
![image](https://github.com/PX4/PX4-user_guide/assets/26798987/1f5de8aa-be55-4427-8a0b-3dc5cbfd6b5b)

@bkueng can you enlighten us? In https://docs.px4.io/main/en/config/actuators.html#reversing-motors it says there is a "Set Spin Direction " button, where is that?

Reviewer 2: > That seems very unintuitive to me

It's a bit of a compromise for an efficient UI layout. I see that point though. Maybe adding 'Permanently' is enough?

> It feels wrong because this will change the directions of all ESCs presumably

No, you can set them individually (there's a dropdown list when clicking the button).

> you can set the direction of CAN or DSHOT ESC through the UI but not PWM.

Only DShot, CAN would still need to be implemented.

> You should ESC calibrate for PWM and One SHot but not Dshot and Can

Yes

> For all ESC you still need to set disarmed, armed, and high values.

It depends. For dshot you don't need to set them, but there's DSHOT_MIN if needed.

Author: Thanks very much @bkueng - I'll adjust the text appropriately.

>>It feels wrong because this will change the directions of all ESCs presumably
> No, you can set them individually (there's a dropdown list when clicking the button).

I don't think this is ever going to be intuitive, but it might be convenient. if it were me I'd have toggle button next to each affected motor - the UI would have to be special though in that the button would hold not have visible - you press it, that changes the state, but it isn't a radio button or similar that shows selected/not selected.

Author: Modifying THIS text to be correct. Will separately update actuator config. @sfuhrer FYI

```suggestion
The final step is [Motor Configuration](../config/actuators.md#motor-configuration):
- [Reverse any motors](../config/actuators.md#reversing-motors) that don't match the spin direction configured in the Geometry.
  For DShot ESC you can do this through the Acuator Testing UI.
- PWM, OneShot, and CAN ESC, set the motor input limits for disarmed, low and high speed (not needed for DShot ESC)
```

Relevant topics:

- [Vehicle (Frame) Selection](../config/airframe.md) - Select vehicle type to match your frame.
- [Actuator Configuration and Testing](../config/actuators.md) - Vehicle geometry, output mapping, motor configuration, testing.
- [ESC Calibration](../advanced_config/esc_calibration.md) - Do between output mapping and motor configuration (topic above) for PWM and OneShot ESC.

## Sensor Setup and Calibration

> PX4 most commonly relies on a magnetometer (compass) for direction information, a barometer for altitude, a gyroscope for attitude, an accelerometer for velocity calculations, and a GPS/GNSS for global position.

Reviewer: ```suggestion
PX4 most commonly relies on a magnetometer (compass) for direction information, a barometer for altitude, a gyroscope for body rates, an accelerometer for attitude and a GPS/GNSS for global position.
```

Pixhawk flight controllers (and many others) have inbuilt magnetometer, accelerometer, gyroscope, and barometer.

> The inbuilt compass usually isn't particularly reliable, and it is common to also add an external compass (usually with a GPS for position).

Reviewer: ```suggestion
The inbuilt compass usually isn't particularly reliable, and it is common to also add an external compass (usually combined with a GNSS receiver in the same device).
```

We first need to set the [Sensor Orientation](../config/flight_controller_orientation.md), informing PX4 how the autopilot (and its inbuilt sensors) and external compasses are oriented relative to the vehicle.
Generally you'll orient towards the front of the vehicle and not have to set anything.
Once that is done we need to calibrate the compass(es), gyroscope, and accelerometer.

The core sensor setup is covered in these topics.

- [Sensor Orientation](../config/flight_controller_orientation.md)
- [Compass](../config/compass.md)
- [Gyroscope](../config/gyroscope.md)
- [Accelerometer](../config/accelerometer.md)

PX4 can use other peripherals, such as distance sensors, optical flow sensors, traffic avoidance alarms, cameras, and so on.

- [Flight Controller Peripherals](../peripherals/README.md) - Setup specific sensors, optional sensors, actuators, and so on.

:::note

> Barometers don't need calibration, and if you've mounted the flight controller level, there isn't much value in a [Level Horizon](../config/level_horizon_calibration.md) calibration either.
> 
> Obviously you don't need to calibrate or configure anything that isn't present or used by PX4.
> For multicopter, that includes [airspeed sensors](../config/airspeed.md), which are not used for flight control.

Author: ```suggestion
Sensors that you don't need to calibrate/configure include:

- [Level Horizon](../config/level_horizon_calibration.md) calibration isn't usually needed if you have mounted the flight controller level.
- Sensors that are not present, or that are not used by PX4 multicopter for flight control, such as [Airspeed sensors](../config/airspeed.md).
- Sensors that don't need calibration, including: Barometers and GNSS.
```

:::

> ## GPS Setup

Reviewer: I don't think the GPS requires a dedicated section.

Author: Agreed. I'm going to remove this. I will include it above as "a sensor that is not covered"

> ## GPS Setup
> 
> <!-- TBD
> 
> Note that it is plugnplay except for RTK. Worth saying.
> 
> -->

Author: ```suggestion
```
